### PR TITLE
test: Unit tests for route visualizer component

### PR DESCRIPTION
**Pull request by OmkarJ13**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=eHvyzfa)
### <samp>🤖 Generated by Copilot at 70d5146</samp>

This pull request improves the testing and linting of the `route-visualizer` component, which shows a map of the user's mileage route or location. It adds and updates test cases in `route-visualizer.component.spec.ts`, using mock data from `mileage-location.data.ts`. It also fixes a linting error in `route-visualizer.component.ts` by removing the optional chaining operator.

### <samp>🤖 Generated by Copilot at 70d5146</samp>

> _We'll mock the data for the `route-visualizer`_
> _And test it with the `gmaps` and `location` provider_
> _We'll fix the linting errors and the optional chaining_
> _And make the map display the mileage or the current bearing_

### <samp>🤖 Generated by Copilot at 70d5146</samp>

*  Add mock data for mileage locations with a null value ([link](https://github.com/fylein/fyle-mobile-app/pull/2087/files?diff=unified&w=0#diff-1700e8ecd31b695c136a5143758787db0567656e2e08511505fb30601d42ad65R86-R98))
*  Update imports and declarations for route visualizer component spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2087/files?diff=unified&w=0#diff-cd34879418743e5fa8e811c07bfd9f9b07db088d96127ce28c0d0ed9b7544760L1-R18))
*  Add constants and variables for mock data and static map urls in spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2087/files?diff=unified&w=0#diff-cd34879418743e5fa8e811c07bfd9f9b07db088d96127ce28c0d0ed9b7544760L29-R49))
*  Modify spy objects for location and gmaps services in spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2087/files?diff=unified&w=0#diff-cd34879418743e5fa8e811c07bfd9f9b07db088d96127ce28c0d0ed9b7544760L36-R61))
*  Modify beforeEach block to set up expectations and calls for spy methods and mock google object in spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2087/files?diff=unified&w=0#diff-cd34879418743e5fa8e811c07bfd9f9b07db088d96127ce28c0d0ed9b7544760L57-R105))
*  Modify and add test cases for route visualizer component methods and scenarios in spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2087/files?diff=unified&w=0#diff-cd34879418743e5fa8e811c07bfd9f9b07db088d96127ce28c0d0ed9b7544760L67-R280))
*  Remove optional chaining operator from current location assignment in `route-visualizer.component.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2087/files?diff=unified&w=0#diff-b40c7d882dae9c3cff13038276ec92375b905681316be9c21e6a7ddb44df455aL66-R67))

## Clickup
https://app.clickup.com/t/85zt429fn

## Code Coverage
image

### CODE REVIEW

File: src/app/shared/components/route-visualizer/route-visualizer.component.ts [hunk 1]
```diff
@@ -86,16 +86,16 @@ export class RouteVisualizerComponent implements OnChanges, OnInit {
...
+    this.directions$ = null;
+    this.directionsMapUrl$ = null;
   }
 
   private renderMap(mileageRoute: MileageRoute) {
     this.directions$ = this.gmapsService.getDirections(mileageRoute);
 
     if (!this.loadDynamicMap) {
       this.directionsMapUrl$ = this.directions$.pipe(
-        filter((directionsResults) => directionsResults?.routes?.length > 0),
```
Comment (OmkarJ13) [inline src/app/shared/components/route-visualizer/route-visualizer.component.ts:98]: This is not needed

File: src/app/core/mock-data/directions-results.data.ts [hunk 2]
```diff
@@ -0,0 +1,964 @@
+export const directionsResults1: any = {
```
Comment (Dimple16) [inline src/app/core/mock-data/directions-results.data.ts:1]: add the type, avoid `any`

File: src/app/core/mock-data/directions-results.data.ts [hunk 3]
```diff
@@ -0,0 +1,964 @@
...
+              polyline: {
+                points: 'c{gtBknx}LAXDRHXBXC\\GPOR',
+              },
+              start_location: {
+                lat: 19.2147392,
+                lng: 73.2031007,
+              },
+              travel_mode: 'DRIVING',
+              encoded_lat_lngs: 'c{gtBknx}LAXDRHXBXC\\GPOR',
+              path: [
```
Comment (Dimple16) [inline src/app/core/mock-data/directions-results.data.ts:161]: check if we can cut short the path somehow? maybe 3 items should do

File: src/app/core/mock-data/directions-results.data.ts [hunk 4]
```diff
@@ -0,0 +1,964 @@
...
+              polyline: {
+                points: 'm{gtBeix}LGEc@SQIIGACCEAECGACCUIaAC]AWG_@CIGO',
+              },
+              start_location: {
+                lat: 19.2147877,
+                lng: 73.202269,
+              },
+              travel_mode: 'DRIVING',
+              encoded_lat_lngs: 'm{gtBeix}LGEc@SQIIGACCEAECGACCUIaAC]AWG_@CIGO',
+              path: [
```
Comment (Dimple16) [inline src/app/core/mock-data/directions-results.data.ts:262]: same as above

File: src/app/shared/components/route-visualizer/route-visualizer.component.spec.ts [hunk 5]
```diff
@@ -54,61 +78,168 @@ describe('RouteVisualizerComponent', () => {
...
+      expect(component.directionsMapUrl$).toBeNull();
+    });
+
+    it('should display a current location map when no valid mileage locations are present', () => {
+      component.mileageLocations = [];
+
+      fixture.detectChanges();
+      component.ngOnChanges();
+
+      expect(component.showCurrentLocation).toBe(true);
```
Comment (Dimple16) [inline src/app/shared/components/route-visualizer/route-visualizer.component.spec.ts:171]: use `.toBeTrue()`

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

**Comment by OmkarJ13** [comment]
Updated the PR with new tests based on comments in this PR - https://github.com/fylein/fyle-mobile-app/pull/2085

**Review by yksolanki9 — COMMENTED** [review]
@OmkarJ13 Dimple is already reviewing this, so my review is not needed

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**